app/src/main/java/memphis/myapplication/psync/ConsumerManager.java

```diff
@@ -90,35 +78,26 @@ public void onData(Interest interest, Data data) {
                 if (!RealmRepository.getInstanceForNonUI().getFriend(friendName).isFriend())
                     return;
 
-                SyncData syncData = new SyncData(interestData);
-                String filename = syncData.getFilename();
-                String fileInfoName = FileManager.getFileName(new Name(syncData.getFilename()), FileManager.FILENAME);
+                Metadata metadata = new Metadata(interestData);
+                String filename = metadata.getFilename();
+                String fileInfoName = FileManager.getFileName(new Name(metadata.getFilename()), FileManager.FILENAME);
 
-                String producer = syncData.getFilename().substring(0, syncData.getFilename().indexOf("/file"));
+                String producer = metadata.getFilename().substring(0, metadata.getFilename().indexOf("/file"));
                 RealmRepository realmRepository = RealmRepository.getInstanceForNonUI();
-                Timber.d("isFile:  " + syncData.isFile());
-                realmRepository.saveNewFile(fileInfoName, syncData.isFeed(), syncData.isLocation(), syncData.isFile(), producer);
+                Timber.d("isFile:  " + metadata.isFile());
+                realmRepository.saveNewFile(fileInfoName, metadata.isFeed(), metadata.isLocation(), metadata.isFile(), producer);
                 realmRepository.close();
 
-                if (syncData.isFeed()) {
+                if (metadata.isFeed()) {
                     Timber.d("For feed");
-                    RealmRepository feedLoopRealmRepository = RealmRepository.getInstanceForNonUI();
-                    Timber.d(interest.getName().toUri());
-                    Timber.d("Friend name: " + friendName);
-                    Timber.d(feedLoopRealmRepository.getSymKey(friendName).toString());
-                    new FetchingTask(context, toastData).execute(new FetchingTaskParams(new Interest(new Name(filename)), feedLoopRealmRepository.getSymKey(friendName), true));
+                    new FetchingTask(context, toastData).execute(new FetchingTaskParams(new Interest(new Name(filename)), true));
 
                 } else {
-                    if (syncData.forMe(SharedPrefsManager.getInstance(context).getUsername())) {
+                    BloomFilter bloomFilter = new BloomFilter(metadata.getBloomFilter().get(0).toNumber(), 0.001, metadata.getBloomFilter().get(-1));
```

Comment:
Same as 0.001 comment above - probably best to move it as default value inside BloomFilter and provide various different ctors.